### ISSUE TITLE
test_core_amd.py::test_reduce_layouts fail with rocm-6.1.0-13052

### ISSUE THREAD

**Issue by xiaohuguo2023** [ROCm/triton#397]
With docker image compute-artifactory.amd.com:5000/rocm-plus-docker/framework/compute-rocm-dkms-no-npi-hipclang:13052_ubuntu22.04_py3.10_pytorch_release-2.1_7f1061a

There is one failed unit test:

=================================== FAILURES ===================================
__________________ test_reduce_layouts[1-src_layout1-128-128] __________________

M = 128, N = 128
src_layout = <test_core_amd.MfmaLayout object at 0x7f9a9980d4e0>, axis = 1
device = 'cuda'

    @pytest.mark.parametrize("M, N", shapes)
    @pytest.mark.parametrize("src_layout", layouts)
    @pytest.mark.parametrize("axis", [0, 1])
    def test_reduce_layouts(M, N, src_layout, axis, device='cuda'):
        if torch.version.hip is not None and _get_warp_size() == 64:
            if src_layout.is_transposed and axis == 0:
                pytest.skip("Reduce along axis 0 is not supported in transposed mfma layout")
        rdims_2d = f"1x{N}" if axis == 0 else f"{M}x1"
        rdims_1d = f"{N}" if axis == 0 else f"{M}"
        store_range = "%7" if axis == 0 else "%1"
        blocked = BlockedLayout([1, 1], [32, 1], [4, 1], [0, 1], [1, 1], [1, 1], [0, 1])
        ir = f"""
        #blocked = {blocked}
        #src = {src_layout}
        module attributes {{"triton_gpu.num-ctas" = 1, "triton_gpu.num-warps" = 4 : i32, "triton_gpu.threads-per-warp" = {_get_warp_size()} : i32}} {{
        tt.func public @kernel_0d1d2c3d4c(%arg0: !tt.ptr<f32> {{tt.divisibility = 16 : i32}}, %arg1: i32 {{tt.divisibility = 16 : i32}}, %arg2: !tt.ptr<f32> {{tt.divisibility = 16 : i32}}) {{
            %0 = tt.make_range {{end = {M} : i32, start = 0 : i32}} : tensor<{M}xi32, #triton_gpu.slice<{{dim = 1, parent = #blocked}}>>
            %1 = tt.expand_dims %0 {{axis = 1 : i32}} : (tensor<{M}xi32, #triton_gpu.slice<{{dim = 1, parent = #blocked}}>>) -> tensor<{M}x1xi32, #blocked>
            %2 = tt.splat %arg1 : (i32) -> tensor<{M}x1xi32, #blocked>
            %3 = arith.muli %1, %2 : tensor<{M}x1xi32, #blocked>
            %4 = tt.splat %arg0 : (!tt.ptr<f32>) -> tensor<{M}x1x!tt.ptr<f32>, #blocked>
            %5 = tt.addptr %4, %3 : tensor<{M}x1x!tt.ptr<f32>, #blocked>, tensor<{M}x1xi32, #blocked>
            %6 = tt.make_range {{end = {N} : i32, start = 0 : i32}} : tensor<{N}xi32, #triton_gpu.slice<{{dim = 0, parent = #blocked}}>>
            %7 = tt.expand_dims %6 {{axis = 0 : i32}} : (tensor<{N}xi32, #triton_gpu.slice<{{dim = 0, parent = #blocked}}>>) -> tensor<1x{N}xi32, #blocked>
            %8 = tt.broadcast %5 : (tensor<{M}x1x!tt.ptr<f32>, #blocked>) -> tensor<{M}x{N}x!tt.ptr<f32>, #blocked>
            %9 = tt.broadcast %7 : (tensor<1x{N}xi32, #blocked>) -> tensor<{M}x{N}xi32, #blocked>
            %10 = tt.addptr %8, %9 : tensor<{M}x{N}x!tt.ptr<f32>, #blocked>, tensor<{M}x{N}xi32, #blocked>
            %11 = tt.splat %arg2 : (!tt.ptr<f32>) -> tensor<{rdims_2d}x!tt.ptr<f32>, #blocked>
            %12 = tt.addptr %11, {store_range} : tensor<{rdims_2d}x!tt.ptr<f32>, #blocked>, tensor<{rdims_2d}xi32, #blocked>
            %13 = tt.load %10 {{cache = 1 : i32, evict = 1 : i32, isVolatile = false}} : tensor<{M}x{N}xf32, #blocked>
            %14 = triton_gpu.convert_layout %13 : (tensor<{M}x{N}xf32, #blocked>) -> tensor<{M}x{N}xf32, #src>
            %15 = "tt.reduce"(%14) ({{
            ^bb0(%arg3: f32, %arg4: f32):
              %16 = "triton_gpu.cmpf"(%arg3, %arg4) {{predicate = 2 : i64}} : (f32, f32) -> i1
              %17 = arith.select %16, %arg3, %arg4 : f32
              tt.reduce.return %17 : f32
            }}) {{axis = {axis} : i32}} : (tensor<{M}x{N}xf32, #src>) -> tensor<{rdims_1d}xf32, #triton_gpu.slice<{{dim = {axis}, parent = #src}}>>
            %18 = triton_gpu.convert_layout %15 : (tensor<{rdims_1d}xf32, #triton_gpu.slice<{{dim = {axis}, parent = #src}}>>) -> tensor<{rdims_1d}xf32, #triton_gpu.slice<{{dim = {axis}, parent = #blocked}}>>
            %19 = tt.expand_dims %18 {{axis = {axis} : i32}} : (tensor<{rdims_1d}xf32, #triton_gpu.slice<{{dim = {axis}, parent = #blocked}}>>) -> tensor<{rdims_2d}xf32, #blocked>
            tt.store %12, %19 {{cache = 1 : i32, evict = 1 : i32}} : tensor<{rdims_2d}xf32, #blocked>
            tt.return
        }}
        }}
        """

        import tempfile
        with tempfile.NamedTemporaryFile(mode='w', suffix='.ttgir') as f:
            f.write(ir)
            f.flush()
            kernel = triton.compile(f.name)

        rs = RandomState(17)
        x = rs.randint(0, 4, (M, N)).astype('float32')
        x = (x.view('uint32') & np.uint32(0xffffe000)).view('float32')

        if axis == 0:
            z = np.zeros((1, N)).astype('float32')
        else:
            z = np.zeros((M, 1)).astype('float32')

        x_tri = torch.tensor(x, device=device)
        z_tri = torch.tensor(z, device=device)
        pgm = kernel[(1, 1, 4)](x_tri, x_tri.stride(0), z_tri)

        z_ref = np.max(x, axis=axis, keepdims=True)

>       np.testing.assert_allclose(z_ref, z_tri.cpu().numpy(), rtol=0.01, atol=1e-3)
E       AssertionError:
E       Not equal to tolerance rtol=0.01, atol=0.001
E
E       Mismatched elements: 2 / 128 (1.56%)
E       Max absolute difference: 6.0434
E       Max relative difference: 0.66826636
E        x: array([[3.],
E              [3.],
E              [3.],...
E        y: array([[3.     ],
E              [3.     ],
E              [3.     ],...

python/test/unit/language/test_core_amd.py:3000: AssertionError
=========================== short test summary info ============================
FAILED python/test/unit/language/test_core_amd.py::test_reduce_layouts[1-src_layout1-128-128] - AssertionError:
Not equal to tolerance rtol=0.01, atol=0.001

Mismatched elements: 2 / 128 (1.56%)
Max absolute difference: 6.0434
Max relative difference: 0.66826636
 x: array([[3.],
       [3.],
       [3.],...
 y: array([[3.     ],
       [3.     ],
       [3.     ],...
SKIPPED [40] python/test/unit/language/test_core_amd.py:1250: fp8 data type is not available on hardware
SKIPPED [1] python/test/unit/language/test_core_amd.py:1555: Not supported: memory out of resource.
SKIPPED [62] python/test/unit/language/test_core_amd.py:1697: incompatible non_k_dim == 32 with MN sizes
SKIPPED [60] python/test/unit/language/test_core_amd.py:1684: triton do not generate MFMA instructions for given block size
SKIPPED [1] python/test/unit/language/test_core_amd.py:2095: test_masked_load_shared_memory[bfloat16] is only supported on AMDGPU
SKIPPED [1] python/test/unit/language/test_core_amd.py:2442: math.scalbn has mismatch issues
SKIPPED [3] python/test/unit/language/test_core_amd.py:2752: Skipped
SKIPPED [4] python/test/unit/language/test_core_amd.py:2728: Skipped
SKIPPED [7] python/test/unit/language/test_core_amd.py:2870: Skipped
SKIPPED [8] python/test/unit/language/test_core_amd.py:2940: Reduce along axis 0 is not supported in transposed mfma layout
=========== 1 failed, 6924 passed, 187 skipped in 4543.90s (1:15:43) ===========
